### ISSUE TITLE
aws_ssm_document.ssm-automation-docs: diffs didn't match during apply

### ISSUE THREAD

**Issue by PaulAtkins** [hashicorp/terraform#13764]
### Terraform Version
0.9.0 - 0.9.3

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_ssm_document (Automation)

### Debug Output
https://gist.githubusercontent.com/PaulAtkins/fdeb3a84c373396faab61463106f3335/raw/cbebc5b9d345f696507a23657d0e274437107a10/gistfile1.txt

### Expected Behavior
Existing Automation document should get updated when running apply

### Actual Behavior
Plan works as expected, but error gets thrown when running apply.
It appears that during the diff, the `Old` values for `name` and `content` are empty strings when running apply

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

1. `terraform plan / apply` to create an Automation document
2. Edit the document
3. `terraform plan` works correctly
4. `terraform apply` throws the error

### Important Factoids

- PR #13491 added support for updating SSM Documents, with schema versions >= 2.0, but for Automation documents, the schema version is 0.3 according to the [AWS docs](http://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-ssm-docs.html#document-schemas-features)
> Documents of type Automation must use schema version 0.3

- I've confirmed I can create a new version and update the default version manually using the AWS Console

- I'm using a `template_file` to render the content of the Automation document.

**Comment by jbardin** [comment]
Hi @PaulAtkins,

Sorry this isn't working, I'm not sure exactly what's going on here.

Do you have configuration example I could use to reproduce this? If not the log output from running with `TF_LOG=trace` would help greatly, just make sure to check it for any sensitive information.

**Comment by PaulAtkins** [comment]
Hey @jbardin,

I've just tried to reproduce this again to get the trace logs, and now it's working as expected, which is awesome and also a bit confusing lol

Previously simple changes like adding a space to a `description` item in the Automation json was enough to cause this to happen And the only way I could update the doc was to manually delete it in AWS and remove it from terraform state, before running an `apply`

I've retested with terraform 0.9.3, and all of the below are now working for me:

- Simple updates (adding a space to a description etc)
- Adding / updating `mainSteps`
- Adding / updating multiple Automation docs.

I'm happy for you to close this, and I can create a new issue with trace logs if I'm able to repo the issue again.

Thanks!
